yt/frontends/amrvac/data_structures.py

```diff
@@ -282,3 +285,26 @@ def _set_code_unit_attributes(self):
         setdefaultattr(self, "temperature_unit", temperature_unit)
         setdefaultattr(self, "pressure_unit", pressure_unit)
         setdefaultattr(self, "magnetic_unit", magneticfield_unit)
+
+    def _check_override_units_amrvac(self):
+        # frontend specific method
+        # note: normalisations in AMRVAC have 3 degrees of freedom. The user can specify a unit length and unit
+        # numberdensity, with the third option either a unit temperature OR unit velocity.
+        # If unit temperature is specified then unit velocity will be calculated accordingly, and vice-versa.
+        # AMRVAC does not allow to specify any other normalisation beside those four.
+        # YT does support overriding other normalisations, this method checks for inconsistencies between
+        # supplied 'units_override' items and those allowed by AMRVAC.
+        if not self.units_override:
+            return
+
+        accepted_overrides = ['length_unit', 'numberdensity_unit', 'temperature_unit', 'velocity_unit']
+
+        # note: _override_code_units() has already been called, so self.units_override has been set
+        for unit_override_name in self.units_override:
+            if not unit_override_name in accepted_overrides:
+                raise ValueError('Only length, numberdensity, temperature or velocity are '
+                                 'accepted overrides for amrvac! ({} was supplied)'.format(unit_override_name))
```

Comment:
*This* breaks my use case. Is there no way to force consistency from overrides `{mass, time, length}` ?

Reply:
Aah yes, actually that _is_ possible now I think about it! Right now it’s literally hard-coded, but it’s possible to calculate the correct normalisations _using_ for example mass and time. That would also fit a bit better in yt _and_ is a bit more user-friendly :)

Reply:
admittedly this stuff is hard to design well to suit everyone, but it's healthy that our use cases are different so we at the very least incorporate both of them in the design :)